scripts/tasks/preprocess-api-history.ts

```diff
@@ -0,0 +1,247 @@
+    }
+
+    validHistoryBlocks.push(possibleHistoryBlock);
+  }
+
+  return validHistoryBlocks;
+}
+
+export const preprocessApiHistory = async (startPath: string) => {
+  const schema = path.resolve(startPath, 'api-history.schema.json');
+
+  let validateAgainstSchema: ValidateFunction<ApiHistory> | null = null;
+
+  try {
+    const ajv = new Ajv();
+    const ApiHistorySchemaFile = await readFile(schema, 'utf-8');
+    const ApiHistorySchema = JSON.parse(
+      ApiHistorySchemaFile
+    ) as JSONSchemaType<ApiHistory>;
+    validateAgainstSchema = ajv.compile(ApiHistorySchema);
+  } catch (error) {
+    logger.error(`Error reading API history schema:\n${error}`);
+    return;
+  }
+
+  const files = await getMarkdownFiles(startPath);
+
+  for (const [filePath, content] of files) {
+    if (!content.includes('<!--')) continue;
+    const tree = fromMarkdown(content);
+
+    const possibleHistoryBlocks = findPossibleApiHistoryBlocks(tree as Root);
+    const validHistoryBlocks = findValidApiHistoryBlocks(
+      possibleHistoryBlocks,
+      filePath,
+      validateAgainstSchema
+    );
+
+    if (validHistoryBlocks.length === 0) continue;
+
+    let newContent = content;
+    let fileLengthDifference = 0;
+
+    // Strip HTML comment tags from history blocks
+    for (const validHistoryBlock of validHistoryBlocks) {
+      const apiHistoryRegexMatches =
+        validHistoryBlock.value.match(apiHistoryRegex);
+
+      if (apiHistoryRegexMatches?.length !== 2) {
+        hasWarned = true;
+        logger.warn(
+          `(Skipping block) Error extracting the API history block inside HTML comment in ${logger.green(
+            filePath
+          )}`
+        );
+        continue;
+      }
+
+      const [, historyBlockWithoutTags] = apiHistoryRegexMatches;
+
+      if (
+        validHistoryBlock.position?.start.offset == null ||
+        validHistoryBlock.position?.end.offset == null
+      ) {
+        hasWarned = true;
+        logger.warn(
+          `(Skipping block) Error getting the start and end position of the API history block in ${logger.green(
+            filePath
+          )}`
+        );
+        continue;
+      }
+
+      const start = newContent.substring(
+        0,
+        validHistoryBlock.position.start.offset + fileLengthDifference
+      );
+      const end = newContent.substring(
+        validHistoryBlock.position.end.offset + fileLengthDifference
+      );
+
+      // Stripping the HTML comment tags of a history block will offset the position of the next history block
+      fileLengthDifference +=
+        historyBlockWithoutTags!.length - validHistoryBlock.value.length;
+
+      newContent = start + historyBlockWithoutTags + end;
+    }
+
+    await writeFile(filePath, newContent, 'utf-8');
+  }
+
+  if (hasWarned) {
+    logger.warn(
+      'Some API history blocks were skipped due to errors. ' +
+        'Run "lint-roller-markdown-api-history" from "electron/lint-roller" for more details.'
+    );
```

Comment:
```suggestion
    logger.warn('Some API history blocks were skipped due to errors.');
```

I don't think we need to mention the linter here, given it's removed from the process and is ingesting docs which should have already passed linting.

Reply:
@erickzhao Asked for this in the July 16th meeting, but alright.